updates/upgrader.go

```diff
@@ -128,6 +139,42 @@ func upgradeCoreNotifyActionHandler(n *notifications.Notification) {
 	}
 }
 
+func upgradeHub() error {
+	if hubUpgradeStarted {
+		return nil
+	}
+
+	// check if we can upgrade
+	if spnHubUpdate == nil || spnHubUpdate.UpgradeAvailable() {
+		identifier := "hub/spn-hub" // identifier, use forward slash!
+		if onWindows {
+			identifier += exeExt
+		}
+
+		// get newest spn-hub
+		new, err := GetPlatformFile(identifier)
+		if err != nil {
+			return err
+		}
+		spnHubUpdate = new
+	} else {
+		return nil
+	}
+
+	if info.GetInfo().Version != spnHubUpdate.Version() {
+		// get random delay with up to three hours
+		delayMinutes, err := rng.Number(3 * 60)
+		if err != nil {
+			return err
+		}
+
+		triggerRestart(time.Duration(delayMinutes) * time.Minute)
+		hubUpgradeStarted = true
+	}
```

Comment:
Maybe we should log that somewhere? If I'm watching for an hub upgrade to be downloaded it would be interesting to know when the restart is scheduled.

Reply:
Now logged in `DelayedRestart`.